boards/st/stm32n6570_dk/stm32n6570_dk.dts

```diff
@@ -0,0 +1,7 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "stm32n6570_dk_common.dtsi"
```

Comment:
bit confused why there is a dts including a board specific dtsi when there is no other dts file for this board?

Reply:
It's a provision for later.
The current dts is the standard one, using external flash. After flashing, a boot pin needs to be switched to start the program which poses problem for automatic testing.
We'll add another dts that will use serial boot. It doesn't require switching the boot pin and we will use that on our test bench.
I didn't add it in this PR to try to keep things minimal.

Reply:
well test files should not be here, you can use board extensions for that https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html#board-extensions and have your test files in your own testing repo

Reply:
Common dtsi file removed. Only one dts remains for each board.